src/language_features/range_formatting.rs

```diff
@@ -29,32 +29,30 @@ pub fn text_document_range_formatting(
         meta,
         req_params,
         move |ctx: &mut Context, meta, results| {
-            let result = results.into_iter().flatten().flatten().collect();
```

Comment:
it was not obvious than flatten().flatten() is filtering None cases, so explicit filter_map() is better in my opinion.